test/e2e/operator/deployment_api.go

```diff
@@ -90,6 +91,37 @@ var _ = deploy.DescribeForSome("API", func(d *deploy.Deployment) bool {
 			})
 		}
 
+		It("driver image shall default to operator image", func() {
+			dep := &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"apiVersion": api.SchemeGroupVersion.String(),
+					"kind":       "Deployment",
+					"metadata": map[string]interface{}{
+						"name": "test-deployment-driver-image",
+					},
+					"spec": map[string]interface{}{
+						// NOTE(avalluri): we do not use lvm mode so that
+						// running this test does not pollute the PMEM space
+						"deviceMode": "direct",
+						"driverName": "test-driver-image",
+					},
+				},
+			}
+
+			deployment, err := toDeployment(dep)
+			Expect(err).ShouldNot(HaveOccurred(), "unstructured to deployment conversion")
+
+			operatorDep, err := findOperatorDeployment(f)
+			Expect(err).ShouldNot(HaveOccurred(), "find operator deployment")
+
+			// operator image should be the driver image
+			deployment.Spec.Image = operatorDep.Spec.Template.Spec.Containers[0].Image
+			deploy.CreateDeploymentCR(f, dep)
```

Comment:
I mis-read the test because the placement of this `deployment` change is awkward: I didn't realize that it has nothing to do with the following `deploy.CreateDeploymentCR`. Can you move `deploy.CreateDeploymentCR` and `defer deploy.DeleteDeploymentCR` up such that it follows directly after the `dep` variable creation?

Then the flow of the test is simpler:
- create deployment
- validate deployment

Reply:
Yes, I agree. Will fix.

Reply:
Done.